packages/markers/src/common/marker.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+/*
+* A marker represents meta information for a given uri
+*/
+export interface IMarker<T> {
```

Comment:
Is there a reason to rename `Marker` to `IMarker`? We don't have a rule that interfaces should be started with `I`.

Reply:
@akosyakov  Should we have such a rule however ? 

The reason I was using `I` before was to differentiate the symbol value like: `const Logger = Symbol(Logger)` from the class `Logger` so that it's not ambiguous when injecting with inversify.

But I do agree having `I` in some places and not in other is confusing.  We should clarify when to use it or not at the very least. 

WDYT ?

Reply:
I just saw your comment below now about that there was already a Logger class...
The only problem is that this means you can't have an implementation of the same name as the interface now unless you rename it all... but that can be ok

Reply:
So the rule would be you can omit I for pure Interfaces that won't ever have a class of the same name I guess.

Reply:
Last time then we discussed, it was that we don't have to use `I`. But we did not want to constraint for cases when it is helpful to disambiguate.

Also, in the context of js, a rule like always use `I` then `interface` keyword is not really applicable. We could use `class` keyword to define `ProblemMarker`, nothing will change except one can use `new` operator to create an instance in addition to the object syntax. So switching keywords will require renaming. It sounds fragile.

Reply:
I seem to have lost some of the past discussion sorry, we're just trying to understand a bit better.

So you want to leave the door open to change an interface to a class ? 

Seems to be doing you will have to change more than the name of the class but need to change all classes that implement it to extend rather than implement no ?

Reply:
> I seem to have lost some of the past discussion sorry, we're just trying to understand a bit better.

Sorry, it was discussed offline at the time of defining code conventions. It does not say anything about it on the purpose.

> So you want to leave the door open to change an interface to a class ?

We don't define something which can be implemented here, but a shape of a JSON-object, which should never be implemented. Both could be used to do it in JS.

> Seems to be doing you will have to change more than the name of the class but need to change all classes that implement it to extend rather than implement no ?

No, there are no implementations, only JSON objects.


We could open an issue and continue the discussion there. This PR should be about moving `Marker` and `ProblemMarker` to common and adjusting imports.

The topic of distinguishing real interfaces from JSON-object interfaces troubles me since improper usage, like implementing JSON-object interfaces or using not serializable types as URI, will lead to troubles. Maybe we could have a rule that real interfaces (to be implemented) should start with `I` if it is not then it is JSON-object (should not be implemented, only instantiated). Then it should be enforced with some tslint rule.

Reply:
> We could open an issue and continue the discussion there. This PR should be about moving Marker and ProblemMarker to common and adjusting imports.

I agree, the renaming should not have been in the same commit at the very least.

>We don't define something which can be implemented here, but a shape of a JSON-object, which should never be implemented. Both could be used to do it in JS.

Right I lost that this was indeed just a json object.

>The topic of distinguishing real interfaces from JSON-object interfaces troubles me since improper usage, like implementing JSON-object interfaces or using not serializable types as URI, will lead to troubles. Maybe we could have a rule that real interfaces (to be implemented) should start with I if it is not then it is JSON-object (should not be implemented, only instantiated). Then it should be enforced with some tslint rule

That seems like a good idea me, even if I wonder how tslint would figure this out. I would kinda need to test if toString() works ?

Reply:
I'll create an issue about this

Reply:
See #623

Reply:
See also #624 about the code style

Reply:
Rename the IMarker to Marker as proposed